Schedule_app/models.py

```diff
@@ -6,17 +6,17 @@
 class ScheduledMovies(models.Model):
     start_date = models.DateField()
     end_date = models.DateField()
-    movie = models.ForeignKey(Movie, on_delete=models.SET_NULL, null=True)
+    movie = models.ForeignKey(Movie, on_delete=models.SET_NULL, null=True, related_query_name='moviepass')
 
     def __str__(self):
         return self.movie.title
 
 
 class ScheduleMovieCinema(models.Model):
-    movie = models.ForeignKey(ScheduledMovies, on_delete=models.SET_NULL, null=True)
-    city = models.ForeignKey(City, on_delete=models.SET_NULL, null=True)
-    cinema = models.ForeignKey(Cinema, on_delete=models.SET_NULL, null=True)
-    hall = models.ForeignKey(CinemaHall, on_delete=models.SET_NULL, null=True)
+    movie = models.ForeignKey(ScheduledMovies, on_delete=models.SET_NULL, null=True, related_query_name='movieback')
+    city = models.ForeignKey(City, on_delete=models.SET_NULL, null=True, related_query_name='cityback')
+    cinema = models.ForeignKey(Cinema, on_delete=models.SET_NULL, null=True, related_query_name='cinemaback')
+    hall = models.ForeignKey(CinemaHall, on_delete=models.SET_NULL, null=True, related_query_name='hallback')
     playing = models.DateTimeField(null=True)
```

Comment:
Normally, a cinema is related to a city and a hall to a cinema. We should not be able to schedule a movie in a hall that is not part of a certain cinema or city, right?

Reply:
Make sure to change this also; you don't need so many foreign keys here as some of the are already linked and we can retrieve the city from the cinema and the cinema from the hall as Alex mentioned above